driver-core/src/main/com/mongodb/internal/connection/OperationContext.java

```diff
@@ -41,6 +44,24 @@ public OperationContext(final RequestContext requestContext, final SessionContex
         this(NEXT_ID.incrementAndGet(), requestContext, sessionContext, timeoutContext, serverApi);
     }
 
+
+    public static OperationContext todoOperationContext() {
+        // TODO (CSOT) should be removed; used at locations that require an OC, but which do not yet have one available
+        return nonUserOperationContext(null);
+    }
+
+    public static OperationContext nonUserOperationContext(final MongoClientSettings settings) {
+        // TODO (CSOT) below is placeholder, validate correctness (serverApi/timeoutSettings might
+        // TODO (CSOT) need to be passed in instead)
+        TimeoutSettings timeoutSettings = TimeoutSettings.create(settings);
```

Comment:
This PR does not implement provision of non-user operation contexts (for example, monitoring, etc.), and usages pass in null. I believe how this should be handled would be easier to determine once we have labelled all such locations (by using the this method). If others feel that this is best done in this PR, I do not feel strongly about this, just erring on the side of less work per PR.

Reply:
Ack - good plan